src/Components/CollectorProfile/CollectorProfileArtistsAdd.tsx

```diff
@@ -0,0 +1,273 @@
+import SearchIcon from "@artsy/icons/SearchIcon"
+import {
+  Box,
+  Button,
+  Clickable,
+  LabeledInput,
+  Message,
+  Spinner,
+  Stack,
+  Text,
+  useToasts,
+} from "@artsy/palette"
+import { FC, useRef, useState } from "react"
+import { Environment, graphql } from "react-relay"
+import { extractNodes } from "Utils/extractNodes"
+import { useClientQuery } from "Utils/Hooks/useClientQuery"
+import { CollectorProfileArtistsAddQuery } from "__generated__/CollectorProfileArtistsAddQuery.graphql"
+import { useDebouncedValue } from "Utils/Hooks/useDebounce"
+import CloseIcon from "@artsy/icons/CloseIcon"
+import { EntityHeaderPlaceholder } from "Components/EntityHeaders/EntityHeaderPlaceholder"
+import { useMutation } from "Utils/Hooks/useMutation"
+import {
+  CollectorProfileArtistsAddCreateUserInterestsMutation,
+  UserInterestCategory,
+  UserInterestInterestType,
+} from "__generated__/CollectorProfileArtistsAddCreateUserInterestsMutation.graphql"
+import { CollectorProfileArtistsAddResult } from "Components/CollectorProfile/CollectorProfileArtistsAddResult"
+import { CollectorProfileArtistsAddNewDialog } from "Components/CollectorProfile/CollectorProfileArtistsAddNewDialog"
+
+interface CollectorProfileArtistsAddProps {
+  description?: string
+  onSuccess: () => void
+  onCancel?: () => void
+  /* When used in the inquiry flow we may have to inject a newly authenticated environment */
+  relayEnvironment?: Environment | null
+}
+
+export const CollectorProfileArtistsAdd: FC<CollectorProfileArtistsAddProps> = ({
+  description,
+  onCancel,
+  onSuccess,
+  relayEnvironment,
+}) => {
+  const [query, setQuery] = useState("")
+
+  const { debouncedValue: debouncedQuery } = useDebouncedValue({
+    value: query,
+    delay: 250,
+  })
+
+  const { data, loading } = useClientQuery<CollectorProfileArtistsAddQuery>({
+    query: QUERY,
+    variables: { query: debouncedQuery },
+    skip: debouncedQuery.length === 0,
+  })
+
+  const artists = extractNodes(data?.matchConnection)
+
+  const handleClear = () => {
+    setQuery("")
+    inputRef.current?.focus()
+  }
+
+  const inputRef = useRef<HTMLInputElement | null>(null)
+
+  const [selected, setSelected] = useState<string[]>([])
+
+  const { sendToast } = useToasts()
+
+  const [mode, setMode] = useState<"Idle" | "Adding" | "New">("Idle")
+
+  const { submitMutation } = useMutation<
+    CollectorProfileArtistsAddCreateUserInterestsMutation
+  >({ mutation: MUTATION, relayEnvironment })
```

Comment:
Do we need to pass the relay environment here?

Reply:
Yeah, during inquiry we go from logged out to logged in without refreshing the page. So we have to pass in the newly authenticated environment to do anything with auth.